### PR TITLE
MXRoomSummary: Refactor Last Message Properties

### PR DESCRIPTION
**Pull request by ismailgulek**

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -31,6 +31,9 @@ Changes to be released in next version
  * MXSession: [MXSession decryptEvent:inTimeline:] is deprecated, use [MXSession decryptEvents:inTimeline:onComplete:] instead.
  * MXCrypto: [MXCrypto decryptEvent:inTimeline:] is deprecated, use [MXCrypto decryptEvents:inTimeline:onComplete:] instead.
  * MXCrypto: [MXCrypto hasKeysToDecryptEvent:] is now asynchronous.
+ * MXRoomSummary: lastMessageEventId, lastMessageOriginServerTs, isLastMessageEncrypted, lastMessageString, lastMessageAttributedString, lastMessageOthers properties removed (vector-im/element-ios/issues/4360).
```
Comment (manuroe) [inline CHANGES.rst:34]: We can remove this line now. Those properties have just been moved and it is explained in the line.
We should have a line saying that `MXRoomSummary.lastMessageEvent` is gone for performance reasons.

File: MatrixSDK/Data/MXRoom.m [hunk 2]
```diff
@@ -2339,11 +2339,16 @@ - (void)removeAllOutgoingMessages
...
-        [self.summary resetLastMessage:nil failure:nil commit:YES];
-    }
+    [mxSession eventWithEventId:self.summary.lastMessage.eventId
+                         inRoom:_roomId
+                        success:^(MXEvent *event) {
+        if (event.sentState != MXEventSentStateSent)
+        {
+            [self.summary resetLastMessage:nil failure:nil commit:YES];
+        }
+    } failure:^(NSError *error) {
```
Comment (manuroe) [inline MatrixSDK/Data/MXRoom.m:2349]: you can use nil. `failure` is nullable (in the code).

File: MatrixSDK/Data/MXRoomLastMessage.m [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+
+- (instancetype)initWithEvent:(MXEvent *)event
+{
+    if (self = [super init])
+    {
+        _eventId = event.eventId;
+        _originServerTs = event.originServerTs;
+        _isEncrypted = event.isEncrypted;
+        _sender = event.sender;
+        _others = [NSMutableDictionary new];
```
Comment (manuroe) [inline MatrixSDK/Data/MXRoomLastMessage.m:51]: We can keep it nil. The property is nullable.

File: CHANGES.rst (hunk 1 above)
Comment (ismailgulek) [reply CHANGES.rst:34]: 👍

File: MatrixSDK/Data/MXRoom.m (hunk 2 above)
Comment (ismailgulek) [reply MatrixSDK/Data/MXRoom.m:2349]: 👍

File: MatrixSDK/Data/MXRoomLastMessage.m (hunk 3 above)
Comment (ismailgulek) [reply MatrixSDK/Data/MXRoomLastMessage.m]: Updated, some usage will be updated as well.

**Review by manuroe — APPROVED** [review]

**Pull request merged**